source/_extensions/controls_js_sim/__init__.py

```diff
@@ -25,74 +25,104 @@
-                        )
-                        outf.write("\n")
-                        outf.write(inf.read())
-                        outf.write("\n")
-
-    return outputFile
+STATIC_DIR = Path(__file__).parent / "_static"
+OUTPUT_FILE = STATIC_DIR / "pid-tune.js"
 
 
-def setup(app: Sphinx) -> Dict[str, Any]:
-    print("Generating and adding controls javascript...")
+def get_source_files():
+    """Get the list of source JavaScript files to be merged and minified."""
 
-    # Perform controls js setup
-    static_dir = Path(__file__).parent / "_static"
+    js_files = []
+    js_root = Path(__file__).parent
+
+    for folder in FOLDER_INCS:
+        # find all js files in the specific folder
+        folder_path = js_root / folder
+        # sort file names alphabetically
+        # this allows a within-folder sort by number prefix if needed.
+        in_file_names = sorted(folder_path.glob("*.js"))
+        js_files.extend(in_file_names)
+
+    return js_files
+
+
+def should_rebuild():
+    """Check if JavaScript needs to be rebuilt based on source file timestamps."""
+
+    # If output doesn't exist, must rebuild
+    if not OUTPUT_FILE.exists():
+        return True
+
+    output_mtime = OUTPUT_FILE.stat().st_mtime
+    source_files = get_source_files()
+
+    # Check if any source file is newer than output
+    for source_file in source_files:
+        if source_file.stat().st_mtime > output_mtime:
+            return True
+
+    return False
 
+
+def merge_and_minify():
+    """Merge and minify the JavaScript source files into a single output file."""
+
+    source_files = get_source_files()
+
+    with OUTPUT_FILE.open("w") as outf:
+        for source_file in source_files:
+            with source_file.open("r") as source:
+                if not debugJS:
+                    # Minify each file independently - again, low bar solution for now
+                    minified = jsmin(source.read())
+                    outf.write(minified)
+                    outf.write("\n")
+                else:
+                    # Verbose, no minify, and add debug markers.
+                    outf.write("\n\n\n")
+                    outf.write(
+                        "//*******************************************************\n"
+                    )
+                    outf.write(
+                        "//*******************************************************\n"
+                    )
+                    outf.write("//**    {}\n".format(source_file))
+                    outf.write(
+                        "//*******************************************************\n"
+                    )
+                    outf.write(
+                        "//*******************************************************\n"
+                    )
+                    outf.write("\n")
+                    outf.write(source.read())
+                    outf.write("\n")

```

Comment:
```suggestion
output_text = ""
for source_file in source_files:
    text = ""
    if not debugJS:
        # Minify each file independently - again, low bar solution for now
        minified = jsmin(source_file.read_text())
        text += minified + "\n"
        continue
    
    # Verbose, no minify, and add debug markers.
    text = f"""\n\n\n
    //*******************************************************
    //*******************************************************
    //**    {source_file}
    //*******************************************************
    //*******************************************************
                              
    {source_file.read_text()}
    """
    output_text += textwrap.dedent(indented_string)

OUTPUT_FILE.write_text(output_text)
```
Further testing with `dedent` is needed, but I think it's more concise.